### PR TITLE
Create Module 1 Secure Computer Systems

### PR DESCRIPTION
**Pull request by MatthewCaseres**

### CODE REVIEW

File: secure-computer-systems/02-design-principles.md
Comment (m4ttsch) [inline secure-computer-systems/02-design-principles.md:69]: I have support for [Katex](https://katex.org) in these notes ... you can do something like `$S_A$` and it should render correctly.

File: secure-computer-systems/02-design-principles.md
Comment (m4ttsch) [inline secure-computer-systems/02-design-principles.md:78]: What is this?

File: secure-computer-systems/02-design-principles.md
Comment (m4ttsch) [inline secure-computer-systems/02-design-principles.md:86]: Does this look alright? I don't think you need newlines in between list items.

**Review by m4ttsch — CHANGES_REQUESTED** [review]

**Comment by m4ttsch** [comment]
looks pretty good. I usually include a welcome.md as a landing page to welcome folks to the course, and provide them with textbook information and contact information and stuff. See my other classes. 

Also, you can remove the images from this PR. I will pull them down from the GitHub url and upload them to my CDN.

**Comment by m4ttsch** [comment]
It might be easier to drop the 0X prefix from the filenames unless you have a strong reason for including them. See my comment on your other PR.

File: secure-computer-systems/02-design-principles.md
Comment (MatthewCaseres) [reply secure-computer-systems/02-design-principles.md]: I was experimenting with adding interactivity in a way that doesn't impact how it is rendered on GitHub.

**Comment by MatthewCaseres** [comment]
In the future it might be easier to include all discussion related to notes in the PR for the notes. I got a bit confused.

**Comment by MatthewCaseres** [comment]
> Also, you can remove the images from this PR. I will pull them down from the GitHub URL and upload them to my CDN.

Okay. Would it be acceptable to go live without these changes or that you could make the changes yourself? I am not the most skilled with git.

Also, when I made welcome.md, it made the sidebar disappear. Can you help me understand what happened?

**Comment by m4ttsch** [comment]
You'll need to add the front matter to welcome.md just like you do to the other .md files; that is, you'll need to add `id`, `lecture`, `course` etc.

The image thing we will have to figure out. What will likely happen is that I will give you read/write access to a subdirectory of my google cloud platform storage so that you can upload images yourself.

Once you make these final changes, I think you will be good to go. I will merge in your changes and they should go live. At this point, the expectation is that you continue to add notes for the remainder of the course material, so let me know earlier rather than later if you see something getting in the way of that.

**Comment by MatthewCaseres** [comment]
Okay, so serving images from GitHub works, but CDN is better for performance reasons? Does GitHub not serve images from a CDN? I like having the original files on GitHub. 

I'm pretty motivated to create a full set of notes, but I'll keep you informed if anything comes up.

**Comment by MatthewCaseres** [comment]
oops wrong button

**Comment by m4ttsch** [comment]
The GitHub url that you are currently using is not fronted by a CDN, so the content will not be served to our worldwide audience in an efficient manner. I am fine with you keeping a backup of the images wherever you'd like, but for the sake of user experience, I want content on the site to be delivered in as optimized a way as possible.

**Comment by MatthewCaseres** [comment]
Content is now served from gitcdn.xyz, will this work?

**Comment by m4ttsch** [comment]
im going to make you an account on firebase tomorrow and give you a read/write subdirectory. I'm not familiar with the CDN you mentioned.

**Comment by MatthewCaseres** [comment]
Cool. Before merge let me finish notes for the third week as well so the students have something more up to date when I post about it.

**Comment by m4ttsch** [comment]
Hey Matt, I added you as a user to my omscs-notes project on google cloud platform. You should have received an email. The role I enabled you for allows you to specifically read and write files at the secure-computer-systems subdirectory within my main storage bucket. Let me know if you have any questions/concerns.

**Comment by m4ttsch** [comment]
Get rid of all .DS_Store files, and remove the images from this PR. I gave you access to the bucket on GCP so you could upload them there. Then, update your image paths in the markdown to point to the GCP bucket. Should be good after that.

File: secure-computer-systems/01-getting-started.md
Comment (m4ttsch) [inline secure-computer-systems/01-getting-started.md:20]: For example, this needs to change to

```md
![](https://assets.omscs.io/secure-computer-systems/....
```

**Comment by MatthewCaseres** [comment]
Images served from CDN. I think it might be ready, but I am not all that confident in the quality of my writing. I might want to give it another pass. You set the bar pretty high in terms of quality, hard to match.

**Comment by m4ttsch** [comment]
Everything looks good, but I still need you to delete any that is not a markdown or text file from this PR. I think we can merge in after that

**Comment by MatthewCaseres** [comment]
okay I guess the gitignore on the top level didn't get the DS_store in submodules. Added another gitignore, removed images.

**Review by m4ttsch — APPROVED** [review]
Congratulations sir 🥳

**Pull request merged**
